client/state/initial-state.js

```diff
@@ -24,7 +25,8 @@ export const MAX_AGE = 7 * DAY_IN_HOURS * HOUR_IN_MS;
 function getInitialServerState() {
 	// Bootstrapped state from a server-render
 	if ( typeof window === 'object' && window.initialReduxState ) {
-		return window.initialReduxState;
+		const serverState = reducer( window.initialReduxState, { type: SERVER_DESERIALIZE } );
+		return pick( serverState, Object.keys( window.initialReduxState ) );
```

Comment:
If I don't, server state will override persisted state with default initial state.

@seear which sections of the tree do you see us bootstrapping, and do we need to be more precise?